packages/gatsby-remark-embed-snippet/src/__tests__/index.js

```diff
@@ -375,6 +438,114 @@ describe(`gatsby-remark-embed-snippet`, () => {
 
       expect(transformed).toMatchSnapshot()
     })
+
+    it(`should support hideline-range and hideline-next-line markers`, () => {
+      fs.readFileSync.mockReturnValue(
+        `
+        // hideline-range{1-2}
+        import React from 'react'
+        import { render } from "react-dom"
+
+
+        ReactDOM.render(
+          <div>
+            <ul>
+              <li>Not hidden</li>
+              <li>Not hidden</li>
+              <li>Not hidden</li>
+              // hideline-range{1-3}
+              <li>Hidden</li>
+              <li>Hidden</li>
+              <li>Hidden</li>
+            </ul>
+          </div>,
+          document.getElementById('root')
+        );
+
+
+        // hideline-next-line
+        console.log('Hidden')
+      `
+          .replace(/^ +/gm, ``)
+          .trim()
+      )
+
+      const markdownAST = remark.parse(`\`embed:hello-world.js\``)
+      const transformed = plugin(
+        { markdownAST },
+        {
+          directory: `examples`,
+        }
+      )
+
+      expect(transformed).toMatchSnapshot()
+    })
+
+    it(`should support hideline-range and highlight markers`, () => {
+      fs.readFileSync.mockReturnValue(
+        `
+        // hideline-range{1-2,9-11,16}
+        // highlight-range{6-8}
+        import React from 'react'
+        import { render } from "react-dom"
+        ReactDOM.render(
+          <div>
+            <ul>
+              <li>Not hidden and highlighted</li>
+              <li>Not hidden and highlighted</li>
+              <li>Not hidden and highlighted</li>
+              <li>Hidden</li>
+              <li>Hidden</li>
+              <li>Hidden</li>
+            </ul>
+          </div>,
```

Comment:
in this example it seems like only last `<li>Not hidden and highlighted</li>` (and closing `</ul>` and `</div>`) are highlighted - is this correct behaviour? From content of this "fixture" seems like not?

Reply:
I understand, but the ```highlight-range``` directive does not consider that the lines in the range of the ```hideline-range``` directive are "already hidden". If the directive considered this, I think it would be much more confusing to work with the directives.
In other words, only the directives config comments are being ommitted from line number calcs.

And the test output will be:

```javascript
ReactDOM.render(
    <div>
        <ul>
            <li>Not hidden and highlighted</li>
            <li>Not hidden and highlighted</li>
            <li>Not hidden and highlighted</li>
        </ul>
    </div>,
    document.getElementById('root') // also highlighted
);
```

Reply:
I also updated the [README](https://github.com/washingtonsoares/gatsby/blob/support-hide-lines/packages/gatsby-remark-embed-snippet/README.md#hide-lines) with a consistent example of using the directive ```hideline-range```

Reply:
That's fine if this is desired behaviour. I was looking at snapshot it it seems like it wraps `</ul>` and `</div>` line in `<span class=\"gatsby-highlight-code-line\">` - is that not right?

Reply:
I also updated the [README](https://github.com/washingtonsoares/gatsby/blob/support-hide-lines/packages/gatsby-remark-embed-snippet/README.md#hide-lines) with a consistent example of using the directive ```hideline-range``` :blush:

Reply:
Let me actually just test it locally, maybe I just read snapshot wrong lol

Reply:
![screen shot 2018-10-11 at 16 36 37](https://user-images.githubusercontent.com/419821/46811881-e7e0c280-cd73-11e8-8bd6-a50066bb2cc9.png)

Using almost same code (after changing import :) ) from tests - this is how it looks - first `<li>`s are not highlighted, but closing `</ul>` and `</div>` is - if this is expected we just need to adjust test to highlight `<li>`s